spec/API_specification/array_object.md

```diff
@@ -366,6 +366,49 @@ Evaluates `x1_i & x2_i` for each element `x1_i` of an array instance `x1` with t
 Element-wise results must equal the results returned by the equivalent element-wise function [`bitwise_and(x1, x2)`](elementwise_functions.md#logical_andx1-x2-).
 ```
 
+(method-__dlpack__)=
+### \_\_dlpack\_\_(/, *, stream=None)
+
+Exports the array as a DLPack capsule, for consumption by {ref}`function-from_dlpack`.
+
+#### Parameters
+
+-   **stream**: _Optional\[int\]_
+
+    -   If given, the CUDA or ROCm stream number the consumer will use. Options are:
+        - `None`: producer must assume the legacy default stream (default),
+        - `1`: the legacy default stream,
```

Comment:
Perhaps it would be useful to remove 1 and 2, because they are CUDA specific for now(ROCm does not support 1 and 2.

Just say stream number represented as a python integer(per platform convention) would work

Reply:
I think we still need to say that 0 is explicitly disallowed for CUDA since it's ambiguous though.

Reply:
I agree, it would be great to point out that that fact. thanks @kkraus14

Reply:
Maybe rather than remove things, add something? E.g. make the description:

Stream number, as Python integer, for all device types that supports streams.

Per-device notes:
- `CUDA: 1, 2, 3...`
- `ROCm: 3....`
- anything else, like OpenCL?

and then refer to the DLPack docs (to be written/extended) for more details?

Specs that are too terse are correct, but not all that useful. Given how complex it was to make this converge, I feel like we need to give implementers as much guidance as possible.

Reply:
thanks @rgommers a per device guidance sounds good. i think it is safe to start with CUDA and ROCm and then expand later

Reply:
> I think we still need to say that 0 is explicitly disallowed for CUDA since it's ambiguous though.

@kkraus14 This is what I kept saying above 🙂 

> Per-device notes:
> 
> * `CUDA: 1, 2, 3...`
> * `ROCm: 3....`
> * anything else, like OpenCL?

@rgommers For ROCm, I think it's ok to keep 0, since this is the only way to use the legacy default stream (if so desired).

Reply:
<img width="580" alt="image" src="https://user-images.githubusercontent.com/98330/108257498-2a461580-715f-11eb-846b-dd7a60d2c594.png">